ocaml/testsuite/tests/typing-layouts/basics.ml

```diff
@@ -2331,3 +2331,39 @@ Error: This expression has type t_float64
        But the layout of t_float64 must be a sublayout of value, because
          of the definition of t40 at line 1, characters 0-16.
 |}]
+
+(**********************************************************************)
+(* Test 41: constraints in manifests in mutually recursive typedecls. *)
+
+(* This example must be rejected. *)
+type t1 = string t2 as (_ : immediate)
+and 'a t2 = 'a
+
+[%%expect{|
+Line 2, characters 0-14:
+2 | and 'a t2 = 'a
+    ^^^^^^^^^^^^^^
+Error:
+       The layout of 'a t2 is value, because
+         it instantiates an unannotated type parameter of t2, defaulted to layout value.
```

Comment:
This doesn't look right to me: I don't see an instantiation of this sort.

Really, the error message should mention `string` somewhere, as that's the root of the problem. (Although actually with the conservative approach, maybe this definition would fail even with `int t2` there.)

Reply:
I agree this error message isn't ideal, but it is already issued in other similar circumances, e.g., [here](https://github.com/ocaml-flambda/flambda-backend/blob/4f2310021bcf28bccdada7d6f2048446cd5a60b2/ocaml/testsuite/tests/typing-layouts-float64/basics.ml#L293).

I think it would be a mistake for the error to mention `string` exactly because the same problem occurs with `int` (I've added that as a test).

I've adjusted this to print a message similar to the `check_constraint` message - see below.  (Though I'm also not totally happy with that since in fact this program is genuinely incorrect and can't be fixed with an annotation.)